ladybug/designday.py

```diff
@@ -0,0 +1,1109 @@
+    def file_path(self, ddy_file_path):
+        self._file_path = os.path.normpath(ddy_file_path)
+
+        if not ddy_file_path.lower().endswith('.ddy'):
+            self._file_path = self._file_path + '.ddy'
+
+        self._folder, self._file_name = os.path.split(self.file_path)
+
+    @property
+    def location(self):
+        """Get or set the location."""
+        return self._location
+
+    @location.setter
+    def location(self, data):
+        assert hasattr(data, 'isLocation'), 'Expected' \
+            ' Location type. Got {}'.format(str(data))
+        self._location = data
+
+    @property
+    def design_days(self):
+        """Get or set the design_days."""
+        return self._design_days
+
+    @design_days.setter
+    def design_days(self, data):
+        assert isinstance(data, list), 'Expected' \
+            ' a list of design days. Got {}'.format(str(data))
+        for item in data:
+            assert hasattr(item, 'isDesignDay'), 'Expected' \
+                ' DesignDay type. Got {}'.format(str(item))
+        self._design_days = data
+        for dd in self._design_days:
+            dd.location = self._location
+
+    @property
+    def isDdy(self):
+        """Return True."""
+        return True
+
+    def ToString(self):
+        """Overwrite .NET ToString."""
+        return self.__repr__()
+
+    def __repr__(self):
+        """Ddy object representation."""
+        return "Ddy File - {} [# days: {}]".format(
+            self.location.city, str(len(self._design_days)))
+
+
+# TODO: add a property for hourly horizontal infrared radiation.
+class DesignDay(object):
+    """An object representing design day conditions.
+
+    properties:
+        name: A text string to set the name of the design day
+        day_type: Choose from 'SummerDesignDay', 'WinterDesignDay' or other
+            EnergyPlus days
+        location: Location for the design day
+        dry_bulb_condition: Ladyubug DryBulbCondition object
+        humidity_condition: Ladyubug HumidityCondition object
+        wind_condition: Ladyubug WindCondition object
+        sky_condition: Ladybug SkyCondition object
+    """
+
+    def __init__(self, name, day_type, location, dry_bulb_condition, humidity_condition,
+                 wind_condition, sky_condition):
+        """Initalize the class."""
+        self.name = str(name)
+        self.day_type = day_type
+        self.location = location
+        self.dry_bulb_condition = dry_bulb_condition
+        self.humidity_condition = humidity_condition
+        self.wind_condition = wind_condition
+        self.sky_condition = sky_condition
+
+    @classmethod
+    def from_ep_string(cls, ep_string, location):
+        """Initalize from an EnergyPlus string of a SizingPeriod:DesignDay.
+
+        args:
+            ep_string: A full string representing a SizingPeriod:DesignDay.
+        """
+        # format the object into a list of properties
+        ep_lines = ep_string.split('\n')
```

Comment:
I would use regular expression here. This method can fail.

Reply:
I originally tried doing a regex but I could not come up with anything that was noticeably better or more reliable than these simple operations.  In the end, it seemed like the regex was overkill when all that I wanted to do was split the string and remove certain characters.

I should also note that the splitting and cleaning operations that I'm doing here can't result in a thrown exception and the check after this process makes sure that the data has been interpreted correctly (I'd need this check anyway whether a regex was used or not).

If you still think a regex is better, than I can do it but what I would implement is effectively the same as what I have done here (but with a longer calculation time that the regex uses).

Alternatively, if you can come up with a regex that it better or more reliable, we should definitely use it.

Reply:
I have implemented the regex as we discussed.  I agree that this way is much cleaner and more reliable than before.